### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by elderapo**
Added windows support using this: [https://gist.github.com/rdp/8363580](https://gist.github.com/rdp/8363580)

### CODE REVIEW

File: impl/windows/windows.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	getVolume(function(err, volume) {
+		if (err) return done(err);
+
+		return done(null, volume === 0);
+	});
+};
+
+var setMuted = function(val, done) {
+	val = val ? '0' : '100';
```
Comment (LinusU) [inline impl/windows/windows.js]: Setting the volume to `100` when unmuting seems like a good way to ruin someones speakers 😄 

I believe that there should be a way to do proper mute/unmute?

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,4 @@ results
 node_modules
 
 npm-debug.log
+!adjust_get_current_system_volume_vista_plus.exe
```
Comment (LinusU) [inline .gitignore]: I don't see why this is needed?

Comment (elderapo) [inline .gitignore]: Without it git just ignores this executable. I am not 100% sure but looks like git ignores `*.exe` by default?

File: impl/windows/windows.js (hunk 1 above)
Comment (elderapo) [inline impl/windows/windows.js]: Yeah you are right. Actually `getMuted` doesn't even work when sound is muted (manually - so it's muted but volume is still > 0). I will try to fix that.

File: impl/windows/windows.js [hunk 3]
```diff
@@ -50,15 +66,15 @@ var setVolume = function(val, done) {
 var getMuted = function(done) {
 	done = done || function () {};
 
-	getVolume(function(err, volume) {
+	getVolumeInfo((err, info) => {
```
Comment (LinusU) [inline impl/windows/windows.js]: Arrow functions doesn't work in node 0.x, let's wait a little bit with breaking compatibility :)

Comment (elderapo) [inline impl/windows/windows.js]: Oops, my bad :P

**Comment by kingio** [comment]
You can also turn up the volume in a less binary way:
1. `volumeController.js`:
```js
var shl = new ActiveXObject('WScript.Shell')
argu = WScript.Arguments(0)
ticks = WScript.Arguments(1)
var code, i

code = (argu === 'up' ? '0xAF' : '0xAE')

for (i = 0; i < ticks; i++) {
    shl.SendKeys(String.fromCharCode(code))
}
```

2. CMD -> `cscript volumeController.js up|down 1|2|3...`
Second argument is how much clicks you wanna increase. Ex: (1) 58->59 or (2) 58->60, etc..

**Comment by elderapo** [comment]
It would be cool to remove `adjust_get_current_system_volume_vista_plus.exe` dependency but I don't think it's possible to achieve that using only VBScript.

First of all for me (tested only on win10) `shl.SendKeys(String.fromCharCode('0xAF'))` turns up volume by 2 (ex. 58 -> 60) which is a little bit weird and I have no idea why it's like that. Same thing happens with lowering the volume level.

Second of all I didn't find a way to get current volume level, mute state using VBScript.

Third of all (lets forget about that 1 instead of 2 increment/decrement volume level bug atm) I didn't find a simple way of setting specific volume level using VBScript. All I was able to do was to set volume level to 0 by executing `shl.SendKeys(String.fromCharCode('0xAE'))` 100 times (so it always gets to lowest volume level possible - 0) and then by executing `shl.SendKeys(String.fromCharCode('0xAF'))` x times (where x is target volume level ex. 33). Also this method is slow. Running `shl.SendKeys(...)` 100+ times takes like 1-2 seconds.

**Comment by kingio** [comment]
The x2 volume up is a windows default setting. Try changing the volume by using your keyboard hotkeys. They will go +2 / -2
I've found another problem with the vbscript.. It only works if explorer.exe is present. Looks like microsoft likes to add multiple non related features to single binaries. (as explorer.exe also holds the background wallpaper)

File: impl/windows/index.js [hunk 4]
```diff
@@ -0,0 +1 @@
+module.exports = require('./windows');
```
Comment (LinusU) [inline impl/windows/index.js]: Could we remove this file and rename `impl/windows/windows.js` to `impl/windows/index.js`?

File: .gitignore (hunk 2 above)
Comment (LinusU) [reply .gitignore]: This is probably due to your global gitignore, but we can keep it in 👍

**Comment by LinusU** [comment]
It would be awesome to have it without binary, but I think that we could still merge this approach for now, and then I would be super happy to accept a pull request that changes it to non-binary :)

**Comment by LinusU** [comment]
Hmm, so for some reason one of the backends at VirusTotal flags this file:

https://www.virustotal.com/#/file/9f0e5f9a6600d28a687460a46828c068e5a00c6e43fc4f6a6453509a72e6ddb4/detection

I'm not sure if it means anything but it makes me a bit hesitant to merge 🤔 

Any ideas?

**Comment by elderapo** [comment]
I just compiled [source code](https://gist.github.com/rdp/8363580) and virustotal.com gives me exact [same result](https://l.facebook.com/l.php?u=https%3A%2F%2Fwww.virustotal.com%2F%23%2Ffile%2Feed345ec00501415d75491c681655253d2dac1a28b4743b1d202740932b4064f%2Fdetection&h=ATMjgKKhWZZv5I0JA3wzbj43s3KAS843R_g9GaVNp9dMfTA8ZQO3YRXAVBGAs9hCXcdQ5lqTKgpPOxcFe8k9fczDGrRTgXFV34mhCdL7N7t_-K1V43y8wZjotXKCfOQ_HuHVmg) so it has to be false positive.

If you want I can try to wrap source code of that exe into node addon and then:
a) Every windows user will need python (or whatever is required by node-gyp) to compile it (automatically) from source code after installation of node-loudness package.
b) Use [node-pre-gyp](https://github.com/mapbox/node-pre-gyp) to precompile binaries for deployment.

**Comment by LinusU** [comment]
Cool, let's merge!

Thank you for your hard work on this 👍

**Pull request merged**